proto/proto/store/block_producer.proto

```diff
@@ -0,0 +1,162 @@
+
+    // Retrieves block header by given block number. Optionally, it also returns the MMR path
+    // and current chain length to authenticate the block's inclusion.
+    rpc GetBlockHeaderByNumber(shared.BlockHeaderByNumberRequest) returns (shared.BlockHeaderByNumberResponse) {}
+
+    // Returns data required to prove the next block.
+    rpc GetBlockInputs(GetBlockInputsRequest) returns (BlockInputs) {}
+
+    // Returns the inputs for a transaction batch.
+    rpc GetBatchInputs(GetBatchInputsRequest) returns (BatchInputs) {}
+
+    // Returns data required to validate a new transaction.
+    rpc GetTransactionInputs(GetTransactionInputsRequest) returns (TransactionInputs) {}
+}
+
+// GET BLOCK INPUTS
+// ================================================================================================
+
+// Returns data required to prove the next block.
+message GetBlockInputsRequest {
+    // IDs of all accounts updated in the proposed block for which to retrieve account witnesses.
+    repeated account.AccountId account_ids = 1;
+
+    // Nullifiers of all notes consumed by the block for which to retrieve witnesses.
+    //
+    // Due to note erasure it will generally not be possible to know the exact set of nullifiers
+    // a block will create, unless we pre-execute note erasure. So in practice, this set of
+    // nullifiers will be the set of nullifiers of all proven batches in the block, which is a
+    // superset of the nullifiers the block may create.
+    //
+    // However, if it is known that a certain note will be erased, it would not be necessary to
+    // provide a nullifier witness for it.
+    repeated primitives.Digest nullifiers = 2;
+
+    // Array of note IDs for which to retrieve note inclusion proofs, **if they exist in the store**.
+    repeated primitives.Digest unauthenticated_notes = 3;
+
+    // Array of block numbers referenced by all batches in the block.
+    repeated fixed32 reference_blocks = 4;
+}
+
+// Represents the result of getting block inputs.
+message BlockInputs {
+    // The latest block header.
+    blockchain.BlockHeader latest_block_header = 1;
+
+    // Proof of each requested unauthenticated note's inclusion in a block, **if it existed in
+    // the store**.
+    repeated note.NoteInclusionInBlockProof unauthenticated_note_proofs = 2;
+
+    // The serialized chain MMR which includes proofs for all blocks referenced by the
+    // above note inclusion proofs as well as proofs for inclusion of the requested blocks
+    // referenced by the batches in the block.
+    bytes partial_block_chain = 3;
+
+    // The state commitments of the requested accounts and their authentication paths.
+    repeated account.AccountWitness account_witnesses = 4;
+
+    // The requested nullifiers and their authentication paths.
+    repeated NullifierWitness nullifier_witnesses = 5;
+}
+
+// A nullifier returned as a response to the `GetBlockInputs`.
+message NullifierWitness {
```

Comment:
Should this message be "nested" in the `BlockInputs`? (to follow the pattern we use in other places).